### PR TITLE
AutoLeave totem check

### PR DESCRIPTION
**Pull request by ThisTestUser**
Adds a slider to indicate the number of totems the player should have before disconnecting. This feature is off by default.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new setting to AutoLeave that allows users to set a threshold for totems before AutoLeave activates.

### CODE REVIEW

**Comment by ThisTestUser** [comment]
Still relevant. It may not be optimal to leave if the player is able to escape death through totems.

File: src/main/java/net/wurstclient/hacks/AutoLeaveHack.java [hunk 1]
```diff
@@ -86,6 +96,10 @@ public void onUpdate()
 		if(currentHealth <= 0F || currentHealth > health.getValueF() * 2F)
 			return;
 		
+		// check totems
+		if(totems.getValueI() != 11 && countTotems(MC.player.getInventory()) > totems.getValueI())
+			return;
```
[the remark this answers is not in the document]
Comment (ThisTestUser) [reply src/main/java/net/wurstclient/hacks/AutoLeaveHack.java]: The magic number is from the setting "totems", where setting the slider to 11 disables the check.

File: src/main/java/net/wurstclient/hacks/AutoLeaveHack.java [hunk 2]
```diff
@@ -116,6 +130,21 @@
...
+		
+		for(int slot = 0; slot <= 36; slot++)
+			if(inventory.getStack(slot).getItem() == Items.TOTEM_OF_UNDYING)
+				totems++;
+		
+		ItemStack offhandStack = inventory.getStack(40);
+		if(offhandStack.getItem() == Items.TOTEM_OF_UNDYING)
+			totems++;
+		
+		return totems;
```
[the remark this answers is not in the document]
Comment (ThisTestUser) [reply src/main/java/net/wurstclient/hacks/AutoLeaveHack.java]: `inventory.getStack(40)` appears in other places as well, for example AutoTotemHack.java.

File: src/main/java/net/wurstclient/hacks/AutoLeaveHack.java [hunk 3]
```diff
@@ -116,6 +131,21 @@ public void onUpdate()
...
+		for(int slot = 0; slot <= 36; slot++)
+			if(inventory.getStack(slot).getItem() == Items.TOTEM_OF_UNDYING)
+				totems++;
+			
+		ItemStack offhandStack = inventory.getStack(40);
+		if(offhandStack.getItem() == Items.TOTEM_OF_UNDYING)
+			totems++;
+		
+		return totems;
+	}
```
[the remark this answers is not in the document]
Comment (Alexander01998) [reply src/main/java/net/wurstclient/hacks/AutoLeaveHack.java:147]: Minecraft has a helper method for getting the offhand stack. I'll make sure we use it.

**Review by Alexander01998 — APPROVED** [review]

**Comment by Alexander01998** [comment]
@coderabbitai review

**Pull request merged**